applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1033,6 +1133,87 @@ static void discover_cb(struct bt_conn *conn, int err, enum bt_audio_dir dir)
 					return;
 				}
 			}
+
+			LOG_DBG("Found %d valid cap(s), %d location(s) and we have %d ep(s)",
+				POPCOUNT(valid_sink_caps), POPCOUNT(unicast_server->location),
+				unicast_server->num_sink_eps);
+
+			memcpy(&unicast_server->sink_codec_cfg, &lc3_preset_sink.codec_cfg,
+			       sizeof(struct bt_audio_codec_cfg));
+			unicast_server->sink_ep = unicast_server->spare_sink_eps[0];
+
+			if (POPCOUNT(unicast_server->location) == 1 &&
+			    POPCOUNT(valid_sink_caps) == 1 && unicast_server->num_sink_eps == 1) {
+
+				ret = bt_audio_codec_cfg_set_val(
+					&unicast_server->sink_codec_cfg,
+					BT_AUDIO_CODEC_CFG_CHAN_ALLOC,
+					(uint8_t *)&unicast_server->location,
+					sizeof(unicast_server->location));
+				if (ret < 0) {
+					LOG_ERR("Failed to set codec channel allocation: %d", ret);
+					return;
+				}
+
+			} else if (POPCOUNT(unicast_server->location) == 2 &&
+				   POPCOUNT(valid_sink_caps) >= 1 &&
+				   unicast_server->num_sink_eps == 2) {
+				/* Stereo usecase */
+				uint32_t left = 1;
+				uint32_t right = 2;
+
+				ret = bt_audio_codec_cfg_set_val(&unicast_server->sink_codec_cfg,
+								 BT_AUDIO_CODEC_CFG_CHAN_ALLOC,
+								 (uint8_t *)&left,
+								 sizeof(unicast_server->location));
+				if (ret < 0) {
+					LOG_ERR("Failed to set codec channel allocation: %d", ret);
+					return;
+				}
+
+				unicast_server->ch_name = "LEFT";
+				unicast_server->location = BT_AUDIO_LOCATION_FRONT_LEFT;
```

Comment:
Something is weird here.

So if `unicast_server->location` contains 2 bits (LEFT | RIGHT), then we then set the `ch_name` and `unicast_server->location` to LEFT and then later set it to RIGHT? 

You also overwrite the pointer to the unicast server in
```c
				unicast_server = &unicast_servers[stereo_idx.lvl1][stereo_idx.lvl2]
								 [stereo_idx.lvl3];
```

Reply:
The clue is in the print on line 1185 `("Creating multiple")`, we first set one unicast_server-instance to left and then clone that with most of the same setting, but we set it to right